src/main/java/net/snowflake/client/jdbc/SnowflakeRichResultSetSerializableV1.java

```diff
@@ -0,0 +1,223 @@
+package net.snowflake.client.jdbc;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import net.snowflake.client.core.QueryResultFormat;
+import net.snowflake.client.core.SFBaseSession;
+import net.snowflake.client.core.SFBaseStatement;
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.log.ArgSupplier;
+import net.snowflake.client.log.SFLogger;
+import net.snowflake.client.log.SFLoggerFactory;
+
+@SnowflakeJdbcInternalApi
+public class SnowflakeRichResultSetSerializableV1 extends SnowflakeResultSetSerializableV1 {
+
+  private static final SFLogger logger =
+      SFLoggerFactory.getLogger(SnowflakeRichResultSetSerializableV1.class);
+
+  private String richResultsFirstChunkStringData;
+  private int richResultsFirstChunkRowCount;
+  private int richResultsChunkFileCount;
+  private int richResultsColumnCount;
+  private final List<ChunkFileMetadata> richResultsChunkFilesMetadata = new ArrayList<>();
+  private byte[] richResultsFirstChunkByteData;
+  private String richResultsQrmk;
+  private final Map<String, String> richResultsChunkHeadersMap = new HashMap<>();
+  private final List<SnowflakeRichResultsColumnMetadata> richResultsColumnMetadata =
+      new ArrayList<>();
+  private QueryResultFormat richResultsQueryResultFormat;
+
+  transient JsonNode richResultFirstChunkRowset = null;
+
+  /**
+   * A factory function for internal usage only It creates SnowflakeRichResultSetSerializableV1 with
+   * NoOpChunksDownloader which disables chunks prefetch
+   */
+  public static SnowflakeRichResultSetSerializableV1 createWithChunksPrefetchDisabled(
+      JsonNode rootNode,
+      SFBaseSession sfSession,
+      SFBaseStatement sfStatement,
+      ResultStreamProvider resultStreamProvider)
+      throws SnowflakeSQLException {
+    return new SnowflakeRichResultSetSerializableV1(
+        rootNode, sfSession, sfStatement, resultStreamProvider, true);
+  }
+
+  private SnowflakeRichResultSetSerializableV1(
+      JsonNode rootNode,
+      SFBaseSession sfSession,
+      SFBaseStatement sfStatement,
+      ResultStreamProvider resultStreamProvider,
+      boolean disableChunksPrefetch)
+      throws SnowflakeSQLException {
+    super(rootNode, sfSession, sfStatement, resultStreamProvider, disableChunksPrefetch);
+    if (!rootNode.at("/richResult").isMissingNode()) {
```

Comment:
Yes, `JsonNode.at()` has slightly different argument format since you can pass there an absolute path to node as a string. Also this method seemed more suitable for checking if it exists in response